cranelift/codegen/src/ir/dfg.rs

```diff
@@ -536,27 +536,26 @@ impl DataFlowGraph {
             "Replacing {} with itself would create a loop",
             dest_inst
         );
+
+        let dest_results = self.results[dest_inst].as_slice(&self.value_lists);
+        let src_results = self.results[src_inst].as_slice(&self.value_lists);
+
         debug_assert_eq!(
-            self.results[dest_inst].len(&self.value_lists),
-            self.results[src_inst].len(&self.value_lists),
+            dest_results.len(),
+            src_results.len(),
             "Replacing {} with {} would produce a different number of results.",
             dest_inst,
             src_inst
         );
 
-        for (&dest, &src) in self.results[dest_inst]
-            .as_slice(&self.value_lists)
-            .iter()
-            .zip(self.results[src_inst].as_slice(&self.value_lists))
-        {
-            let original = src;
+        for (&dest, &original) in dest_results.iter().zip(src_results) {
```

Comment:
The bound `(dest, original)` names are slightly confusing given we're iterating over `zip(dest_results, src_results)` -- can we pick either `original` or `src` and use it in both places?